anyio/_backends/_trio.py

```diff
@@ -240,12 +240,19 @@ async def wait(self):
         await self._event.wait()
 
 
-class Condition(trio.Condition):
+class Condition:
```

Comment:
Why is this change needed? If somebody is expecting anyio's Condition to be type compatible with trio's, this change breaks that assumption.

Reply:
Because subclassing trio.Condition now triggers a warning.
That expectation would fail anyway whenever a backend other than `trio` is used.

Reply:
We deprecated inheriting from most of trio's public types. More details: https://github.com/python-trio/trio/issues/1044